packages/jsii/lib/jsii-diagnostic.ts

```diff
@@ -0,0 +1,791 @@
+      code: 9003,
+      formatter: (location: string) =>
+        `Unable to resolve module location "${location}"`,
+      name: 'miscellaneous/unresolveable-module',
+    });
+
+    public static readonly JSII_9004_UNABLE_TO_COMPUTE_SIGNATURE = Code.error({
+      code: 9004,
+      formatter: (methodName: string, type: spec.Type) =>
+        `Unable to compute signature for method "${methodName}" of "${type.fqn}"`,
+      name: 'miscellaneous/unable-to-compute-signature',
+    });
+
+    public static readonly JSII_9998_UNSUPORTED_NODE = Code.message({
+      code: 9998,
+      formatter: (kindOrMessage: ts.SyntaxKind | string) =>
+        typeof kindOrMessage === 'string'
+          ? kindOrMessage
+          : `Unsupported ${ts.SyntaxKind[kindOrMessage]} node. This declaration will not be accessible from other languages.`,
+      name: 'miscellaneous/unsupported-node',
+    });
+
+    public static readonly JSII_9999_RELATED_INFO = Code.suggestion({
+      code: 9999,
+      formatter: (str: string) => str,
+      name: 'miscellaneous/related-info',
+    });
+
+    //////////////////////////////////////////////////////////////////////////////
+    private static message<T extends DiagnosticMessageFormatter>({
+      code,
+      name,
+      formatter,
+    }: {
+      code: number;
+      formatter: T;
+      name: string;
+    }) {
+      return new Code<T>(code, name, ts.DiagnosticCategory.Message, formatter);
+    }
+
+    private static suggestion<T extends DiagnosticMessageFormatter>({
+      code,
+      name,
+      formatter,
+    }: {
+      code: number;
+      formatter: T;
+      name: string;
+    }) {
+      return new Code<T>(
+        code,
+        name,
+        ts.DiagnosticCategory.Suggestion,
+        formatter,
+      );
+    }
+
+    private static warning<T extends DiagnosticMessageFormatter>({
+      code,
+      name,
+      formatter,
+    }: {
+      code: number;
+      formatter: T;
+      name: string;
+    }) {
+      return new Code<T>(code, name, ts.DiagnosticCategory.Warning, formatter);
+    }
+
+    private static error<T extends DiagnosticMessageFormatter>({
+      code,
+      name,
+      formatter,
+    }: {
+      code: number;
+      formatter: T;
+      name: string;
+    }) {
+      return new Code<T>(code, name, ts.DiagnosticCategory.Error, formatter);
+    }
+
+    /**
+     * Get a diagnostic code by code or name.
+     *
+     * @param codeOrName the looked up diagnostic code or name.
+     *
+     * @returns the JsiiDiagnosticCode instande, if one exists, or `undefined`
+     */
+    public static lookup(codeOrName: string | number): Code | undefined {
+      if (typeof codeOrName === 'number') {
+        return this.byCode[codeOrName];
+      }
+      return this.byName[codeOrName];
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
+    readonly #defaultCategory: ts.DiagnosticCategory;
```

Comment:
This `#` is throwing me off and I can't quickly find what it means. Do you have a reference?

Reply:
https://github.com/tc39/proposal-class-fields/blob/master/PRIVATE_SYNTAX_FAQ.md